components/display/ILI9341.c

```diff
@@ -10,356 +11,392 @@
-//ML 	Vertical Refresh Order 		LCD vertical refresh direction control.
-//BGR 	RGB-BGR Order 				Color selector switch control
-//									(0=RGB color filter panel, 1=BGR color filter panel)
-//MH 	Horizontal Refresh ORDER 	LCD horizontal refreshing direction control.
-// Bits 17-0
-//		XX XX XX XX XX XX XX XX XX XX MY MX MV ML BGR MH 0 0
-typedef enum  {
-	MAC_BIT_MH=2,
-	MAC_BIT_BGR,
-	MAC_BIT_ML,
-	MAC_BIT_MV,
-	MAC_BIT_MX,
-	MAC_BIT_MY,
-} mac_bits;
-
-uint16_t set_mac_bit(mac_bits bit, uint16_t val){
-	return (1 << bit) | val;
-}
-uint16_t unset_mac_bit(mac_bits bit, uint16_t val){
-	return ~(1 << bit) & val;
-}
-
 static void SetLayout( struct GDS_Device* Device, bool HFlip, bool VFlip, bool Rotate ) { 
 	struct PrivateSpace *Private = (struct PrivateSpace*) Device->Private;
-	Private->ReMap = HFlip ? (Private->ReMap & ~(1 << MAC_BIT_MX)) : (Private->ReMap | (1 << MAC_BIT_MX));
-	Private->ReMap = VFlip ? (Private->ReMap | (1 << MAC_BIT_MY)) : (Private->ReMap & ~(1 << MAC_BIT_MY));
-	Device->WriteCommand( Device, L1_CMD_MEMORY_ACCESS_CONTROL );
-	Device->WriteData( Device, &Private->ReMap, 1 );
-	WriteDataByte(Device,0x00);
+	ESP_LOGI(TAG, "SetLayout 197 HFlip=%d VFlip=%d Rotate=%d (1=true)", HFlip, VFlip, Rotate);
+	//        D/CX RDX WRX D17-8 D7 D6 D5 D4 D3  D2 D1 D0 HEX
+	//Command   0   1   ↑    XX  0  0  1  1  0   1  1  0  36h
+	//Parameter 1   1   ↑    XX  MY MX MV ML BGR MH 0  0  00
+	//Orientation 0: MADCtl = 0x80  =   1000 0000 (MY=1)
+	if ((Device->Height)>(Device->Width)){		//Resolution = 320x240
+		Private->MADCtl = (1 << 7);				// 0x80 = default (no Rotation an no Flip)
+		if (HFlip) {							//Flip Horizontal
+			int a = Private->MADCtl;
+			Private->MADCtl = (a ^ (1 << 7));
+		}
+		if (Rotate) {							//Rotate 180 degr.
+			int a = Private->MADCtl;
+			a = (a ^ (1 << 7));
+			Private->MADCtl = (a ^ (1 << 6));
+		}
+		if (VFlip) {							//Flip Vertical
+			int a = Private->MADCtl;
+			Private->MADCtl = (a ^ (1 << 6));
+		}
+	} else {									//Resolution = 240x320
+		Private->MADCtl = (1 << 5);				// 0x20 = default (no Rotation an no Flip)
+		if (HFlip) {							//Flip Horizontal
+			int a = Private->MADCtl;
+			Private->MADCtl = (a ^ (1 << 6));
+		}
+		if (Rotate) {							//Rotate 180 degr.
+			int a = Private->MADCtl;
+			a = (a ^ (1 << 7));
+			Private->MADCtl = (a ^ (1 << 6));
+		}
+		if (VFlip) {							//Flip Vertical
+			int a = Private->MADCtl;
+			Private->MADCtl = (a ^ (1 << 7));
+		}
+	}
+
+	/*//----- Or Rotation: -----
+	Private->MADCtl = 0x80;				//Orientation 0 degree
+	if (HFlip) {						//Orientation 90 degree
+		Private->MADCtl = 0x20;
+		int a = Device->Height;
+		int b = Device->Width;
+		Device->Height = b;
+		Device->Width = a;
+	}
+	if (Rotate) {						//Orientation 180 degree
+		Private->MADCtl = 0x40;
+	}
+	if (VFlip) {						//Orientation 270 degree
+		Private->MADCtl = 0xE0;
+		int a = Device->Height;
+		int b = Device->Width;
+		Device->Height = b;
+		Device->Width = a;
+	}
+	*/
+
```

Comment:
Do we need to keep these comments? I'm always trying to remove dead code unless we really want to keep it